### DISCUSSION TITLE
Meta query error: Type loader is expected to return a callable or valid type

### DISCUSSION THREAD

**Discussion by taylorhlabs** [wp-graphql/wp-graphql#2152] [Q&A]
I'm trying to pass an array as a variable to a meta query but I get the following error: **Type loader is expected to return a callable or valid type \"RootQueryToAccommodationConnectionWhereArgsMetaArray\", but it returned null**

I have the following query set up in GraphiQL IDE. Note that the query works fine when I hard code the values in place of the $metaArray variable.
```
query MyQuery($metaArray: [RootQueryToAccommodationConnectionWhereArgsMetaArray] = {compare: EQUAL_TO, key: "is_active", value: "yes"}) {
  accommodations(where: {metaQuery: {relation: AND, metaArray: $metaArray}}) {
    nodes {
      accommodationDetails {
        numberDiamonds
        numberStars
      }
      title
    }
  }
}
```

**Comment by wilrevehl** [comment]
I have the exact same issue. Did you ever figure out the solution @taylorhlabs ? 

@jasonbahl sorry to bug ya, but what's the trick to getting list_of type arrays into input args? The WhereArgs are so abstracted this has been quite a mystery to solve. Thanks!

**Reply by wilrevehl** [reply]
@jasonbahl I was able to resolve this by adding a custom scalar to meta. I noticed your taxonomy plugin includes the ability to inject a key value pair as a variable array so I'll plan to look and see how you registered that one. I'm guessing I can go down the type inheritance a level or two, maybe using a custom interface or something?

**Reply by wilrevehl** [reply]
In my case, it ultimately turned out to be something very subtle about your hooks. There's a function that sequences when to insert custom fields into the registry **get_graphql_register_action()**: 

```add_action( get_graphql_register_action(), [ $this, 'register_types' ], 10, 1 );```

Once I switched to using that, the custom input fields stop throwing the error above.
